sdk/core/Azure.Core/src/Serialization/SerializationExtensions.cs

```diff
@@ -0,0 +1,88 @@
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// Extensions that can be used for serialization.
+    /// </summary>
+    public static class SerializationExtensions
+    {
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to the specified type using
+        /// the provided <see cref="ObjectSerializer"/>.
+        /// </summary>
+        /// <typeparam name="T">The type that the data should be
+        /// converted to.</typeparam>
+        /// <param name="data">The <see cref="BinaryData"/> instance to convert.</param>
+        /// <param name="serializer">The serializer to use
+        /// when deserializing the data.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use during deserialization.</param>
+        ///<returns>The data converted to the specified type.</returns>
+        public static T ToObject<T>(this BinaryData data, ObjectSerializer serializer, CancellationToken cancellationToken = default) =>
+            (T)serializer.Deserialize(data.ToStream(), typeof(T), cancellationToken);
+
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to the specified type using
+        /// the provided <see cref="ObjectSerializer"/>.
+        /// </summary>
+        /// <typeparam name="T">The type that the data should be
+        /// converted to.</typeparam>
+        /// <param name="data">The <see cref="BinaryData"/> instance to convert.</param>
+        /// <param name="serializer">The serializer to use
+        /// when deserializing the data.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use during deserialization.</param>
+        ///<returns>The data converted to the specified type.</returns>
+        public static async ValueTask<T> ToObjectAsync<T>(this BinaryData data, ObjectSerializer serializer, CancellationToken cancellationToken = default) =>
+            (T)await serializer.DeserializeAsync(data.ToStream(), typeof(T), cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Convert the provided value to it's binary representation and return it as a <see cref="BinaryData"/> instance.
+        /// </summary>
+        /// <param name="serializer">The serializer to use when creating the <see cref="BinaryData"/> instance.</param>
+        /// <param name="value">The value to convert.</param>
+        /// <param name="inputType">The type to use when serializing <paramref name="value"/>. If omitted, the type will be determined using <see cref="object.GetType"/>().</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use during serialization.</param>
+        /// <returns>The object's binary representation as <see cref="BinaryData"/>.</returns>
+        public static BinaryData SerializeToBinaryData(this ObjectSerializer serializer, object? value, Type? inputType = default, CancellationToken cancellationToken = default)
```

Comment:
We can implement this method more efficiently if it's a virtual on the ObjectSerializer

Reply:
What's the benefit of having it as an extension method?

Reply:
Good point - there doesn't seem to be a practical different for users in having this be an extension method. However, due to the suggestion to use the Azure namespace here, there would be a difference as ObjectSerializer lives in the Azure.Core.Serialization namespace.

Reply:
Leaving all the extension methods in Azure namespace.